symex-evil.el

```diff
@@ -49,6 +49,53 @@
   :enable (normal)
   :exit-hook (symex-exit-mode))
 
+(defun symex-evil-repeat-pre-hook-advice (&rest _)
+  "Prepare the current command for recording the repeation.
+
+This function is meant to advise `evil-repeat-pre-hook' which
+starts recording a repeation during the `pre-command-hook', but
+only records a repition when in normal or visual state. This
+calls `evil-repeat-start' if the buffer is currently in symex
+state."
+  (when evil-local-mode
+    (let ((repeat-type (evil-repeat-type this-command t)))
+      (when (and (not (evil-repeat-force-abort-p repeat-type))
+                 (not (or (null repeat-type)
+                          (evil-mouse-events-p (this-command-keys))))
+                 (evil-symex-state-p))
+        (evil-repeat-start)))))
+(put 'symex-evil-repeat-pre-hook-advice 'permanent-local-hook t)
```

Comment:
I'm not familiar with `permanent-local-hook`, but is it standard practice to do this declaration at the top level instead of in an initialization function, e.g. one that could be called from `symex-evil-initialize`?

Reply:
Now that I'm thinking about it more, we probably don't need it at all, sense these functions are advising other hooks rather than being used as hook functions themselves. The manual says to use this property to prevent a function from being cleared out of a hook variable when the major mode switches:

> If the value is non-nil, the named function should not be deleted from the local value of a hook variable when changing major modes.

But that likely doesn't apply to advice.

I put these lines here like this though, just because that's how evil-repeat.el does it for me its hook functions.

I'll try to experiment a little more with and without this property.